test/Either/Either.js

```diff
@@ -1,46 +1,172 @@
 'use strict';
 
 var jsc = require('jsverify');
+var R = require('ramda');
 
 var S = require('../..');
 
-var Compose = require('../internal/Compose');
 var EitherArb = require('../internal/EitherArb');
 var Identity = require('../internal/Identity');
 var IdentityArb = require('../internal/IdentityArb');
+var laws = require('../internal/laws');
+var squareRoot = require('../internal/squareRoot');
 
 
 suite('Either', function() {
 
+  suite('Setoid laws', function() {
+
+    var setoidLaws = laws.Setoid;
+
+    setoidLaws.reflexivity(
+      EitherArb(jsc.string, jsc.falsy)
+    );
+
+    setoidLaws.symmetry(
+      EitherArb(jsc.bool, jsc.bool),
+      EitherArb(jsc.bool, jsc.bool)
+    );
+
+    setoidLaws.transitivity(
+      EitherArb(jsc.bool, jsc.bool),
+      EitherArb(jsc.bool, jsc.bool),
+      EitherArb(jsc.bool, jsc.bool)
+    );
+
+  });
+
+  suite('Semigroup laws', function() {
+
+    var semigroupLaws = laws.Semigroup(R.equals);
+
+    semigroupLaws.associativity(
+      EitherArb(jsc.string, jsc.string),
+      EitherArb(jsc.string, jsc.string),
+      EitherArb(jsc.string, jsc.string)
+    );
+
+  });
+
+  suite('Functor laws', function() {
+
+    var functorLaws = laws.Functor(R.equals);
+
+    functorLaws.identity(
+      EitherArb(jsc.string, jsc.number)
+    );
+
+    functorLaws.composition(
+      EitherArb(jsc.string, jsc.number),
+      jsc.constant(Math.sqrt),
```

Comment:
I saw `squareRoot` used in couple places and when I saw `Math.sqrt`, though that it was a mistake.